client/components/ReviewForm.js

```diff
@@ -0,0 +1,89 @@
+import React, { Component } from 'react'
+import { Redirect } from 'react-router-dom';
+import axios from 'axios'
+import { minReviewLength, maxReviewLength } from '../../server/constants'
+
+export default class ReviewForm extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      rating: 0,
+      text: '',
+      artwork: {},
+      error: '',
+      redirect: false
+    }
+    this.handleChange = this.handleChange.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.starChange = this.starChange.bind(this)
+  }
+
+  handleChange(ev) {
+    this.setState({...this.state, [ev.target.name]: ev.target.value})
+  }
+
+  starChange(rating) {
+    this.setState({...this.state, rating})
+  }
+
+  async componentDidMount() {
+    const artwork = (await axios.get(`/api/artworks/${this.props.match.params.id}`)).data
+    this.setState({...this.state, artwork})
+  }
+
+  async handleSubmit(ev) {
+    ev.preventDefault()
+    if(this.state.text.length < minReviewLength || this.state.text.length > maxReviewLength) this.setState({...this.state, error: `Review must be between ${minReviewLength} and ${maxReviewLength} characters.`})
+    else if(this.state.rating === 0) this.setState({...this.state, error:"Choose a star rating 1 to 5."})
+    else {
+      try {
+        const review = (await axios.post('/api/reviews', { ...this.state })).data
+        if(review) this.setState({...this.state, redirect: true})
+      } catch(err) {
+        this.setState({...this.state, error: `Review was not created.`})
+        console.log(err)
+      }
+    }
+  }
+
+  render() {
+    const { artwork, rating, text, error, redirect } = this.state
+    if (redirect) {
+      return <Redirect to={`/artworks/${artwork.id}`}/>;
+    }
+    const starRatings=[]
+    for (let i = 1; i <= 5; i++) {
+      starRatings.push(<option key={i} value={i}>{i}</option>)
+    }
+    return (
+      <div>
+        {artwork.id ?
+          <form id="login-form" onSubmit={this.handleSubmit}>
```

Comment:
hm, this is something we should change in the beautifying round, but id's should only be used one unique component. if we want to reuse the styling for the login form then we need to change it to a class and make the name something more generic like "submission-form"